task-launcher/src/tasks/shared/helpers/fetchAndParseCorpus.js

```diff
@@ -66,34 +64,26 @@ const transformCSV = (csvInput, numOfPracticeTrials, sequentialStimulus) => {
       newRow.distractors = newRow.distractors.map((choice) => camelize(choice));
     }
 
-    totalTrials += 1;
-
-    if (row.notes === 'practice') {
-      totalPractice += 1;
-    }
-
     let currentTrialType = newRow.trialType;
-
-    // console.log('currentTrialType:', currentTrialType)
-    // console.log('currTrialTypeBlock:', currTrialTypeBlock)
-
     if (currentTrialType !== currTrialTypeBlock) {
       currTrialTypeBlock = currentTrialType;
       currPracticeAmount = 0;
     }
 
-    // Only push in the specified amount of practice trials
-    if (newRow.notes !== 'practice') {
+    if (newRow.notes === 'instructions') {
       stimulusData.push(newRow);
+      totalTrials += 1
     } else if (newRow.notes === 'practice' && currPracticeAmount < numOfPracticeTrials) {
+      // Only push in the specified amount of practice trials
       stimulusData.push(newRow);
       currPracticeAmount += 1;
+      totalTrials += 1
+    } else if (newRow.notes !== 'practice') {
+      stimulusData.push(newRow);
+      totalTrials += 1
```

Comment:
This code can be more concise:
```suggestion
    if (newRow.notes === 'practice' && currPracticeAmount < numOfPracticeTrials) {
      currPracticeAmount += 1;
    }
    
    stimulusData.push(newRow);
    totalTrials += 1
```

Reply:
stimulusData.push() should not be done when currPracticeAmount exceeds numOfPracticeTrials.
when we skip a practice trial we don't want to push or increase totalTrials

Reply:
I updated the suggestion

Reply:
Doing this instead:
```
    if (newRow.notes === 'practice') { 
      if (currPracticeAmount < numOfPracticeTrials) {
        // Only push in the specified amount of practice trials
        currPracticeAmount += 1;
        stimulusData.push(newRow);
        totalTrials += 1
```suggestion
    if (newRow.notes === 'instructions') {
      stimulusData.push(newRow);
      totalTrials += 1
    } else if (newRow.notes === 'practice' && currPracticeAmount < numOfPracticeTrials) {
      // Only push in the specified amount of practice trials
      stimulusData.push(newRow);
      currPracticeAmount += 1;
      totalTrials += 1
    } else if (newRow.notes !== 'practice') {
      stimulusData.push(newRow);
      totalTrials += 1
```
      } // else skip extra practice
    } else {
      // instruction and stimulus
      stimulusData.push(newRow);
      totalTrials += 1
    }
```